### PR TITLE
WIP: ArgoCD Bootstrap

### PR DESCRIPTION
**Pull request by tylerauerbeck**
This adds an initial bootstrapping of ArgoCD to the bootstrap deployment. Instructions are included in the README and you are able to tell the bootstrap to just deploy the applications or Argo+Applications.

What this does:
- Provides way of installing ArgoCD as part of Bootstrap 
- Allows you to specify initial ArgoCD password
- Allows you to deploy ArgoCD + App of Apps together
- Allows you to specify list of known-hosts to add to ArgoCD

I'm thinking that OpenShift Auth + RBAC will be done as a follow up PR, but let me know if there are any other thoughts there.

cc/ @pabrahamsson @oybed @jacobsee

### CODE REVIEW

File: bootstrap/values.yaml [hunk 1]
```diff
@@ -5,4 +5,29 @@ application:
   name: open-management-portal
   url: https://github.com/rht-labs/open-management-portal-deployment.git
   path: applications/
-  ref: master
\ No newline at end of file
+  ref: master
+
+argo-cd:
+  enabled: false
+  installCRDs: false
```
Comment (jacobsee) [inline bootstrap/values.yaml:12]: Does this mean that we need to install the `Application` CRD elsewhere? Or is this telling Argo whether or not it should install CRDs it encounters later?

Comment (tylerauerbeck) [reply bootstrap/values.yaml:12]: Nope, this is just a key that allows for helm v2 compatability. Long story short, they changed the way that CRD's were handled between v2 and v3 and this is how this particular set of charts is handling that.

Comment (jacobsee) [reply bootstrap/values.yaml:12]: 👍

File: bootstrap/values.yaml [hunk 2]
```diff
@@ -5,4 +5,29 @@ application:
...
-  ref: master
\ No newline at end of file
+  ref: master
+
+argo-cd:
+  enabled: false
+  installCRDs: false 
+
+  global:
+# Do we want this?
```
Comment (oybed) [inline bootstrap/values.yaml:15]: The short answer is no, but let's discuss

Comment (oybed) [reply bootstrap/values.yaml:15]: Discussed and agreed that this is the approach for now. Hoping for no Argo bugs and good PR/MR process/automation to control content being submitted in "management repos" 

@pabrahamsson would like to follow up with you on the latter - i.e.: process/automation control of repo content.

**Comment by tylerauerbeck** [comment]
Moving this back to WIP while we evaluate another approach.

**Comment by tylerauerbeck** [comment]
Closing in favor of #44

**Pull request closed without merging**